chains/btc/listener/util.go

```diff
@@ -6,15 +6,18 @@ import (
 
 	"github.com/ChainSafe/sygma-relayer/chains/btc/config"
 	"github.com/btcsuite/btcd/btcjson"
+	"github.com/btcsuite/btcd/btcutil"
 )
 
 const (
 	WitnessV1Taproot = "witness_v1_taproot"
 	OP_RETURN        = "nulldata"
 )
 
-func DecodeDepositEvent(evt btcjson.TxRawResult, resource config.Resource) (Deposit, bool, error) {
+func DecodeDepositEvent(evt btcjson.TxRawResult, resource config.Resource, feeAddress btcutil.Address) (Deposit, bool, error) {
```

Comment:
Btw how do we handle when there might be multiple deposit events inside one transaction?

Reply:
With this implementation, it won't work. The last OP_RETURN evm address will get all the deposits. If we have a standard of how the deposit looks in terms of order of output data eg. 1 output amount, second fee, third destination. Then we could handle multiple deposits in one transaction. I can open an issue for that, so we can handle this in a separate PR

Reply:
Let's make an issue for this to track it

Reply:
Here is the [issue](https://app.zenhub.com/workspaces/sygma-protocol-62f126a54b7d5a55203732f0/issues/gh/sygmaprotocol/sygma-relayer/302)